Common/NearestNeighbour/src/main/java/org/tribuo/common/nearest/KNNModel.java

```diff
@@ -87,22 +92,28 @@
         INNERTHREADPOOL
     }
 
-    private final Pair<SparseVector,T>[] vectors;
+    private final Pair<SGDVector,T>[] vectors;
```

Comment:
I'm not sure if this is a serialization safe refactoring. I think it should be, but have you tested loading in a 4.2 KNN into this code? And if you've got such a test, might be worth adding it to the unit tests. We've got a few serialization compatibility tests that have been added when the on disk format has changed slightly.

Reply:
This is a good point, and model serialization should certainly be tested, especially for cases like this. I've added a test which a model generated using the old KNN model. Let me know if you there are some ways the test can be improved - the model filename may be a little verbose, etc.